### PR TITLE
Bluetooth: Controller: Replace Central connection spacing Kconfig

### PR DESCRIPTION
**Pull request by cvinayak**
Replace the BT_CTLR_ADVANCED_SCHEDULING_CENTRAL_CONN_SPACING Kconfig with BT_CTLR_CENTRAL_SPACING to configure the the spacing from the beginning of a central connection event to beginning of the next central connection when supporting multiple simulataneous central connections.

If the BT_CTLR_CENTRAL_SPACING in microseconds is smaller then the required time reservation for a connection event, for given maximum data length and PHY in use, then the calculated time reservation will be used as the spacing between consecution central connection events.

The change fixes an issue wherein use of old Kconfig with higher values causes new central connections to be placed in between already connected central role if there was free time space available to fit the new connections time reservation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_sched.c [hunk 1]
```diff
@@ -876,7 +876,10 @@ static struct ull_hdr *ull_hdr_get_cb(uint8_t ticker_id, uint32_t *ticks_slot)
 
 		conn = ll_conn_get(ticker_id - TICKER_ID_CONN_BASE);
 		if (conn && !conn->lll.role) {
-			*ticks_slot = conn->ull.ticks_slot;
+			*ticks_slot =
+				MAX(conn->ull.ticks_slot,
+				    HAL_TICKER_US_TO_TICKS(
+					    CONFIG_BT_CTLR_CENTRAL_SPACING));
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_sched.c:882]: CMakeLists inclusion of `ull_sched.c` depends on `CONFIG_BT_CTLR_SCHED_ADVANCED`, hence a guard is not required in this file.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c [hunk 2]
```diff
@@ -460,7 +460,7 @@ uint32_t lll_preempt_calc(struct ull_hdr *ull, uint8_t ticker_id,
 	uint32_t diff;
 
 	ticks_now = ticker_ticks_now_get();
-	diff = ticks_now - ticks_at_event;
+	diff = ticker_ticks_diff_get(ticks_now, ticks_at_event);
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c:463]: This change is duplicated in PR #52124, undo this change here

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c:463]: Without this commit, the bsim_bt_multiple test asserts. When #52124 is merged I will rebase this PR.

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/Kconfig.ll_sw_split [hunk 3]
```diff
@@ -440,25 +440,17 @@ config BT_CTLR_SCHED_ADVANCED
 	  Disabling this feature will lead to overlapping role in timespace
 	  leading to skipped events amongst active roles.
 
-config BT_CTLR_SCHED_ADVANCED_CENTRAL_CONN_SPACING
-	int "Advanced scheduling central preferred connection spacing"
+config BT_CTLR_CENTRAL_SPACING
+	int "Central Connection Spacing"
 	depends on BT_CTLR_SCHED_ADVANCED
 	default 0
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/Kconfig.ll_sw_split:446]: added range

**Comment by cvinayak** [comment]
@wopu-ot Could someone from your team review this change, whether this is acceptable?

**Review by mtpr-ot — APPROVED** [review]
LGTM

**Review by kruithofa — APPROVED** [review]
LGTM
